src/app/Providers/components/AddEditProviderModal/AddEditProviderModal.tsx

```diff
@@ -279,14 +316,54 @@ export const AddEditProviderModal: React.FunctionComponent<IAddEditProviderModal
                   />
                 ) : null}
                 {fields?.hostname ? (
-                  <ValidatedTextInput
-                    field={fields.hostname}
-                    label={getLabelName('hostname', brandPrefix(providerType))}
-                    isRequired
-                    fieldId="hostname"
-                  />
+                  <>
+                    <ValidatedTextInput
+                      field={fields.hostname}
+                      label={getLabelName('hostname', brandPrefix(providerType))}
+                      isRequired
+                      fieldId="hostname"
+                      inputProps={{
+                        onFocus: () => setCertificateQueryFlag(false),
+                        onBlur: () => {
+                          setHostname(forms.vsphere.fields.hostname.value);
```

Comment:
I think we can simplify things and reduce the amount of state we're using here.

* I think we should define `isCertificateValid` as a required field in the `vsphere` section of `useAddProviderFormState` rather than in its own `useState`. We may also be able to just remove the `fingerprint` and `fingerprintFilename` fields from there, but we'd need to pass the fingerprint as an additional param to `mutateProvider` (which comes from `useCreateProviderMutation` and `usePatchProviderMutation`).
* By default (without the custom `onBlur` here), the `fields.hostname.isTouched` boolean will not be true until the blur event happens (because that's what triggers the validation errors to appear). Currently the validation on the hostname field is broken here because you're not calling `setIsTouched` in your custom `onBlur`. But I think we can restore the default behavior by removing the custom `onBlur` and `isCertificateQueryEnabled` state entirely, because:
  * If we replace the `setCertificateQueryFlag(false)` in `onFocus` with `fields.hostname.setIsTouched(false)`, we can use `fields.hostname.isTouched` as our value for whether the query should be enabled.
* If we do that, we also don't need the `hostname`/`setHostname` state, because we already have `fields.hostname.value`.

So in short, I think all we really need is the `isCertificateValid` to be part of the form state (so `isFormValid` works properly), and to manage the `isTouched` value of the hostname field as our indicator of whether the query should run, and we can get rid of all the other new state you've introduced here. Everything below that uses `isCertificateValid` can use `fields.isCertificateValid.value`, and your `useCertificateQuery` can take `fields.hostname.isTouched` as an `enabled` param.

Reply:
@mturley, 

Thank you for your thorough review.
With latest commit is as close I could get to the expected behavior. 
The reason I keep using a state hostname is to make benefit of having a button triggering the fetch and avoid trying to fetch the certificate at each key stroke when hostname is entering.